### PR TITLE
Cleanup Autosuggest description and change title for autosuggest and instant results

### LINKED ISSUE #3191: BUG: Autosuggest description when using ElasticPress.io
### Describe the bug

There are some problems in the [Autosuggest::epio_allowed_parameters()](https://github.com/10up/ElasticPress/blob/4.4.0/includes/classes/Feature/Autosuggest/Autosuggest.php#L846) method that we need to address:

- There is no need to compare the WP version anymore, as the minimum supported version is already 5.6
- The link to EP.io is not working (the variable is empty)
- The logo there seems to be misplaced. Perhaps we should pull it to the top of the message, add it as a link, or move it to somewhere else.

Props to @NV607FOX

### Steps to Reproduce

1. Setup ElasticPress using an EP.io account
2. Go to the Features page
3. Open the Autosuggest feature

### Screenshots, screen recording, code snippet

_No response_

### Environment information

_No response_

### WordPress and ElasticPress information

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by burhandodhy**
### Description of the Change

This PR updates the titles for the Autosuggest and Instant Results feature if the ElasticPress is connected with ep.io. Also it removed the old checks from the autosuggest description. 

image

Closes #3191 

### Changelog Entry

> Changed - Update title for Autosuggest and Instant results feature, if connected to ep.io

### Credits

Props @burhandodhy @felipeelia @NV607FOX 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: includes/classes/Feature.php [hunk 1]
```diff
@@ -344,4 +344,15 @@ public function output_settings_box() {
...
 	}
+
+	/**
+	 * Returns the ElasticPress.io logo.
+	 *
+	 * @since 4.4.1
+	 * @return string
+	 */
+	public function get_epio_logo() : string {
+
```
Comment (felipeelia) [inline includes/classes/Feature.php:386]: Open to discussion but can we remove the empty line between the method declaration and its content? I'm seeing this being added lately and it is not something used in WordPress in general.

File: includes/classes/Feature/Autosuggest/Autosuggest.php [hunk 2]
```diff
@@ -939,4 +899,19 @@ public function epio_autosuggest_set_and_get() {
...
+	 * @since 4.4.1
+	 * @return string
+	 */
+	protected function get_title() : string {
+
+		if ( ! Utils\is_epio() ) {
+			return esc_html__( 'Autosuggest', 'elasticpress' );
+		}
+
+		/* translators: 1. Feature name;  */
```
Comment (felipeelia) [inline includes/classes/Feature/Autosuggest/Autosuggest.php]: The placeholder won't be replaced by the feature name but with the EP.io logo

**Review by felipeelia — CHANGES_REQUESTED** [review]
@burhandodhy in addition to my comments below, there is also a visual problem we will need to address. This is how I'm seeing it in my local:

image

File: includes/classes/Feature/Autosuggest/Autosuggest.php (hunk 2 above)
Comment (burhandodhy) [reply includes/classes/Feature/Autosuggest/Autosuggest.php]: Thanks. Its fixed now.

File: includes/classes/Feature.php (hunk 1 above)
Comment (burhandodhy) [reply includes/classes/Feature.php:386]: Makes sense. Done

File: includes/classes/Feature.php [hunk 3]
```diff
@@ -344,4 +344,15 @@ public function output_settings_box() {
...
+		return sprintf( '<img class="feature-epio-logo" src="%s">', esc_url( plugins_url( '/images/logo-elasticpress-io.svg', EP_FILE ) ) );
```
Comment (JakePT) [inline includes/classes/Feature.php]: @burhandodhy Can we add an `alt` attribute here.

**Review by JakePT — CHANGES_REQUESTED** [review]

File: includes/classes/Feature.php (hunk 3 above)
Comment (burhandodhy) [reply includes/classes/Feature.php]: @JakePT added. Can you please give another look?

File: includes/classes/Feature.php [hunk 4]
```diff
@@ -344,4 +344,14 @@ public function output_settings_box() {
...
 	}
+
+	/**
+	 * Returns the ElasticPress.io logo.
+	 *
+	 * @since 4.4.1
+	 * @return string
+	 */
+	public function get_epio_logo() : string {
+		return sprintf( '<img class="feature-epio-logo" alt="elasticpress.io-logo" src="%s">', esc_url( plugins_url( '/images/logo-elasticpress-io.svg', EP_FILE ) ) );
```
Comment (JakePT) [inline includes/classes/Feature.php]: So the alt text should be human readable, so the hyphen isn't necessary here. I would use `ElasticPresss.io logo`. Also, and I should have mentioned this before, ideally images have a width and height attribute. This ensures there's no shifting around during the page load. As far as I can tell this should be `width="110" height="20"`.

**Review by JakePT — CHANGES_REQUESTED** [review]

**Review by JakePT — DISMISSED** [review]

File: includes/classes/Feature/Autosuggest/Autosuggest.php [hunk 5]
```diff
@@ -854,61 +854,21 @@ public function epio_allowed_parameters() {
...
+				printf(
+					/* translators: 1: <a> tag (ElasticPress.io); 2. </a>; 3: <a> tag (KB article); 4. </a>; 5: <a> tag (Site Health Debug Section); 6. </a>; */
+					esc_html__( 'You are directly connected to %1$sElasticPress.io%2$s, ensuring the most performant Autosuggest experience. %3$sLearn more about what this means%4$s or %5$sclick here for debug information%6$s.', 'elasticpress' ),
+					'<a href="' . esc_url( $epio_link ) . '">',
+					'</a>',
+					'<a href="' . esc_url( $epio_autosuggest_kb_link ) . '">',
+					'</a>',
+					'<a href="' . esc_url( admin_url( 'site-health.php?tab=debug' ) ) . '">',
+					'</a>'
+				);
```
Comment (felipeelia) [inline includes/classes/Feature/Autosuggest/Autosuggest.php]: This function call has an extra tab (it should be aligned with the variables.) Also, we should link it to the status report page now (the autosuggest section specifically.)

File: includes/classes/Feature/Autosuggest/Autosuggest.php [hunk 6]
```diff
@@ -939,4 +899,19 @@ public function epio_autosuggest_set_and_get() {
...
 	}
 
+	/**
+	 * Returns the title.
+	 *
+	 * @since 4.4.1
+	 * @return string
+	 */
+	protected function get_title() : string {
+
```
Comment (felipeelia) [inline includes/classes/Feature/Autosuggest/Autosuggest.php]: do you mind removing the extra line here as well @burhandodhy ?

File: includes/classes/Feature/InstantResults/InstantResults.php [hunk 7]
```diff
@@ -992,4 +992,19 @@ public function get_args_schema() {
...
 	}
 
+	/**
+	 * Returns the title.
+	 *
+	 * @since 4.4.1
+	 * @return string
+	 */
+	protected function get_title() : string {
+
```
Comment (felipeelia) [inline includes/classes/Feature/InstantResults/InstantResults.php]: do you mind removing the extra line here as well @burhandodhy ?

**Review by felipeelia — CHANGES_REQUESTED** [review]
@burhandodhy some last changes before we merge it. Also, I don't know if it is something in my computer but the logo looks "pixelated" to me here (see below.) Even the full-size svg looks a bit pixelated to me.
image

File: includes/classes/Feature/InstantResults/InstantResults.php (hunk 7 above)
Comment (burhandodhy) [reply includes/classes/Feature/InstantResults/InstantResults.php]: Done

File: includes/classes/Feature/Autosuggest/Autosuggest.php (hunk 5 above)
Comment (burhandodhy) [reply includes/classes/Feature/Autosuggest/Autosuggest.php]: Done

File: includes/classes/Feature/Autosuggest/Autosuggest.php (hunk 6 above)
Comment (burhandodhy) [reply includes/classes/Feature/Autosuggest/Autosuggest.php]: Done

File: includes/classes/Feature/Autosuggest/Autosuggest.php [hunk 8]
```diff
@@ -854,61 +854,21 @@ public function epio_allowed_parameters() {
...
+			$epio_autosuggest_kb_link = 'https://elasticpress.zendesk.com/hc/en-us/articles/360055402791';
+
+			printf(
+				/* translators: 1: <a> tag (ElasticPress.io); 2. </a>; 3: <a> tag (KB article); 4. </a>; 5: <a> tag (Site Health Debug Section); 6. </a>; */
+				esc_html__( 'You are directly connected to %1$sElasticPress.io%2$s, ensuring the most performant Autosuggest experience. %3$sLearn more about what this means%4$s or %5$sclick here for debug information%6$s.', 'elasticpress' ),
+				'<a href="' . esc_url( $epio_link ) . '">',
+				'</a>',
+				'<a href="' . esc_url( $epio_autosuggest_kb_link ) . '">',
+				'</a>',
+				'<a href="' . esc_url( admin_url( 'site-health.php?tab=debug' ) ) . '">',
```
Comment (felipeelia) [inline includes/classes/Feature/Autosuggest/Autosuggest.php]: @burhandodhy can you please change this link to point to the Status Report page? Thanks.

**Review by felipeelia — CHANGES_REQUESTED** [review]

**Pull request merged**